vault/expiration.go

```diff
@@ -555,13 +555,13 @@ func (m *ExpirationManager) revokeCommon(ctx context.Context, leaseID string, fo
 	}
 
 	// Delete the entry
-	if err := m.deleteEntry(leaseID); err != nil {
+	if err := m.deleteEntry(ctx, leaseID); err != nil {
 		return err
 	}
 
 	// Delete the secondary index, but only if it's a leased secret (not auth)
 	if le.Secret != nil {
-		if err := m.removeIndexByToken(le.ClientToken, le.LeaseID); err != nil {
+		if err := m.removeIndexByToken(ctx, le.ClientToken, le.LeaseID); err != nil {
```

Comment:
Just a note: if we start using physical.Transactional in other locations this is a really good candidate. Ideally we'd delete this and the entry together so either both remain or both go.